api/v1alpha1/argocd_types.go

```diff
@@ -339,6 +339,18 @@ type ArgoCDPrometheusSpec struct {
 	Size *int32 `json:"size,omitempty"`
 }
 
+// Configuration to customize resource behavior (optional) can be configured via splitted sub keys.
+type ResourceCustomizationsNew struct {
```

Comment:
We probably should find a better name than "New" as it will get old pretty quickly.

Reply:
After all, this new struct is not necessary.